src/main/groovy/com/twcable/grabbit/jcr/ProtoNodeDecorator.groovy

```diff
@@ -91,14 +91,22 @@ class ProtoNodeDecorator {
      * @param property
      * @param node
      */
-    private static void addMixins(ProtoPropertyDecorator property, JCRNode node) {
+    private static void addMixins(ProtoPropertyDecorator property, JCRNode node, Session session) {
         property.values.valueList.each { ProtoValue value ->
-            if (node.canAddMixin(value.stringValue)) {
-                node.addMixin(value.stringValue)
-                log.debug "Added mixin ${value.stringValue} for : ${node.name}."
+            try{
+                if(!node.isCheckedOut()){
+                    session.getWorkspace().getVersionManager().checkout(node.path)
+                }
```

Comment:
@sagarsane : I tried finding out a case that would give me this exception but could not find any.. This change was based on what I found in the documentation and checking that things that were working before and after the change. Having a case for this would have helped checking different it from server and client side but 😕